x-pack/plugins/maps/public/routing/routes/maps_app/maps_app_view.js

```diff
@@ -139,145 +131,78 @@ export class MapsAppView extends React.Component {
   };
 
   _updateFromGlobalState = ({ changes, state: globalState }) => {
-    if (!changes || !globalState) {
+    if (!this.state.initialized || !changes || !globalState) {
       return;
     }
-    const newState = {};
-    Object.keys(changes).forEach((key) => {
-      if (changes[key]) {
-        newState[key] = globalState[key];
-      }
-    });
 
-    this.setState(newState, () => {
-      this._appStateManager.setQueryAndFilters({
-        filters: getData().query.filterManager.getAppFilters(),
-      });
-      const { time, filters, refreshInterval } = globalState;
-      this.props.dispatchSetQuery(refreshInterval, filters, this.state.query, time);
-    });
+    this._onQueryChange({ time: globalState.time, refresh: true });
   };
 
-  _getInitialLayersFromUrlParam() {
-    const locationSplit = window.location.href.split('?');
-    if (locationSplit.length <= 1) {
-      return [];
-    }
-    const mapAppParams = new URLSearchParams(locationSplit[1]);
-    if (!mapAppParams.has('initialLayers')) {
-      return [];
-    }
+  async _updateIndexPatterns() {
+    const { nextIndexPatternIds } = this.props;
 
-    try {
-      let mapInitLayers = mapAppParams.get('initialLayers');
-      if (mapInitLayers[mapInitLayers.length - 1] === '#') {
-        mapInitLayers = mapInitLayers.substr(0, mapInitLayers.length - 1);
-      }
-      return rison.decode_array(mapInitLayers);
-    } catch (e) {
-      getToasts().addWarning({
-        title: i18n.translate('xpack.maps.initialLayers.unableToParseTitle', {
-          defaultMessage: `Initial layers not added to map`,
-        }),
-        text: i18n.translate('xpack.maps.initialLayers.unableToParseMessage', {
-          defaultMessage: `Unable to parse contents of 'initialLayers' parameter. Error: {errorMsg}`,
-          values: { errorMsg: e.message },
-        }),
-      });
-      return [];
+    if (nextIndexPatternIds === this._prevIndexPatternIds) {
```

Comment:
I understand you didn't change this comparison logic too much, but while we're in here, shouldn't we be doing a `_.isEqual(nextIndexPatternIds, this._prevIndexPatternIds)` instead?